src/Raven.Server/Documents/Indexes/Configuration/SingleIndexConfiguration.cs

```diff
@@ -20,7 +20,23 @@ public SingleIndexConfiguration(IndexConfiguration clientConfiguration, RavenCon
         {
             _databaseConfiguration = databaseConfiguration;
 
-            Initialize(key => clientConfiguration.GetValue(key) ?? databaseConfiguration.GetSetting(key), throwIfThereIsNoSetMethod: false);
+            Initialize(key =>
+            {
+                var clientValue = clientConfiguration.GetValue(key);
+                if (clientValue != null)
+                    return clientValue;
+
+                if (key == Constants.Configuration.Indexing.MaxMapIndexOutputsPerDocument ||
+                    key == Constants.Configuration.Indexing.MaxMapReduceIndexOutputsPerDocument)
+                {
+                    clientValue = clientConfiguration.GetValue(Constants.Configuration.Indexing.MaxIndexOutputsPerDocument);
```

Comment:
We have 2 real keys:
- MaxMapIndexOutputsPerDocument
- MaxMapReduceIndexOutputsPerDocument

And one 'fake' one used by IndexDefinition:
- MaxIndexOutputsPerDocument

I think this is the best way to handle this, because in IndexDefinition user does not care if this is a Map or Reduce index, so it sets the 'fake' key from which the configuration inherits when it is created.